sdk/src/main/java/io/opentelemetry/sdk/metrics/Aggregator.java

```diff
@@ -30,31 +30,22 @@
   void merge(T aggregator);
 
   /**
-   * LongAggregator represents the base class for all the available aggregations that work with long
-   * values.
+   * Checkpoints this aggregator by saving the previous value and resetting it. The old value is
+   * available to "merge".
    */
-  @ThreadSafe
-  interface LongAggregator<T extends LongAggregator<?>> extends Aggregator<T> {
+  void checkpoint();
 
-    /**
-     * Updates the current aggregator with a newly recorded value.
-     *
-     * @param value the new {@code long} value to be added.
-     */
-    void update(long value);
-  }
+  /**
+   * Updates the current aggregator with a newly recorded long value.
+   *
+   * @param value the new {@code long} value to be added.
+   */
+  void updateLong(long value);
```

Comment:
From your other PR #833, methods `recordLong(long)` and `recordDouble(double)` are introduced with the same Javadoc. Correct me if I am wrong: these two methods are for "Measures" metrics while the methods in the other PR are for "Counters". If so, I would update the doc to make this explicit

Reply:
Done.